features/backup/src/test/java/ir/mehdiyari/krypt/backup/DataViewModelTest.kt

```diff
@@ -0,0 +1,119 @@
+package ir.mehdiyari.krypt.backup
+
+import app.cash.turbine.test
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import ir.mehdiyari.krypt.backup.logic.backup.BackupRepository
+import ir.mehdiyari.krypt.files.logic.repositories.api.FilesRepository
+import ir.mehdiyari.krypt.files.logic.utils.FilesUtilities
+import ir.mehdiyari.krypt.files.logic.utils.MediaStoreManager
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestCoroutineScheduler
+import kotlinx.coroutines.test.TestDispatcher
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+import java.io.File
+
+@OptIn(ExperimentalCoroutinesApi::class)
+internal class DataViewModelTest {
+    @get:Rule
+    val mainDispatcherRule = MainDispatcherRule()
+
+    private val backupRepository = mockk<BackupRepository>(relaxed = true)
+    private val filesRepository = mockk<FilesRepository>(relaxed = true)
+    private val filesUtilities = mockk<FilesUtilities>(relaxed = true)
+    private val mediaStoreManager = mockk<MediaStoreManager>(relaxed = true)
+    private val scheduler = TestCoroutineScheduler()
+    private val dispatcher = UnconfinedTestDispatcher(scheduler)
+
+    private val dataViewModel = DataViewModel(
+        backupRepository,
+        filesRepository,
+        dispatcher,
+        filesUtilities,
+        mediaStoreManager
+    )
+
```

Comment:
Hey Akram, Thanks for this great PR, is good to have `tearDown` method with `@After` annotation, to clear up the mocks. Here and for the rest of the tests.